internal/services/shared/errors.go

```diff
@@ -168,14 +165,6 @@ func rewriteError(ctx context.Context, err error, config *ConfigForErrors) error
 	case errors.As(err, &relationNotFoundError):
 		return spiceerrors.WithCodeAndReason(err, codes.FailedPrecondition, v1.ErrorReason_ERROR_REASON_UNKNOWN_RELATION_OR_PERMISSION)
 
-	case errors.As(err, &maxDepthError):
```

Comment:
FYI `dispatch.MaxDepthExceededError` implements GRPCStatus so it gets caught in line 142 above